src/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs

```diff
@@ -77,6 +77,58 @@ public class SslStreamNetworkStreamTest
             listener.Stop();
         }
 
+        [Fact]
+        public async Task SslStream_SendReceiveOverNetworkStream_AuthenticationException()
+        {
+            TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
+
+            using (X509Certificate2 serverCertificate = Configuration.Certificates.GetServerCertificate())
+            using (TcpClient client = new TcpClient())
+            {
+                listener.Start();
+
+                Task clientConnectTask = client.ConnectAsync(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndpoint).Port);
+                Task<TcpClient> listenerAcceptTask = listener.AcceptTcpClientAsync();
+
+                await Task.WhenAll(clientConnectTask, listenerAcceptTask);
```

Comment:
This shouldn't fail, but you could use the WhenAllOrAnyFailed helper just in case.

Reply:
Good point, to keep in line with the rest of this test project I will use the TestConfiguration.WhenAllOrAnyFailedWithTimeout